be/src/storage/lake/tablet_writer.h

```diff
@@ -143,13 +147,34 @@ class TabletWriter {
 
     const DictColumnsValidMap& global_dict_columns_valid_info() const { return _global_dict_columns_valid_info; }
 
+    void decide_pk_parallel_execution() {
+        if (!config::enable_pk_parallel_execution || _schema->keys_type() != KeysType::PRIMARY_KEYS ||
+            _schema->has_separate_sort_key()) {
+            return;
+        }
+        // For primary key table with single key column and the type is not VARCHAR/CHAR,
+        // we can't enable pk parrallel execution. The reason is that, in the current implementation,
+        // when encoding a single-key column of a non-binary type, big-endian encoding is not used,
+        // which may result in incorrect ordering between sst and segment files.
+        // This is a legacy bug, but for compatibility reasons, it will not be supported in the first phase.
+        // Will fix it later.
+        if (_schema->num_key_columns() > 1 || _schema->column(0).type() == LogicalType::TYPE_VARCHAR ||
+            _schema->column(0).type() == LogicalType::TYPE_CHAR) {
+            _enable_pk_parallel_execution = true;
+        }
+        return;
```

Comment:
return is useless

Reply:
There are 2 PRs remains, I will update these on next PR.